api.go

```diff
@@ -96,25 +97,30 @@ func handle(router *mux.Router, path string, fn http.HandlerFunc, logger *logrus
 			"duration": time.Since(start),
 		}).Info("Request")
 	}
-	router.HandleFunc(path, wrapped)
+	router.HandleFunc(path, wrapped).Methods(methods...)
 }
 
 // NewAPIServer is the constructor for an APIServer
-func NewAPIServer(syncer *Syncer, port uint16, baseLogger *logrus.Entry) {
+func NewAPIServer(syncer *Syncer, port uint16, baseLogger *logrus.Entry, metrics *sqmetrics.SquareMetrics) {
 	logger := baseLogger.WithField("logger", "api_server")
 	apiServer := APIServer{syncer: syncer, logger: logger}
 	router := mux.NewRouter()
 
-	handle(router, "/debug/pprof", pprof.Index, logger)
-	handle(router, "/debug/pprof/cmdline", pprof.Cmdline, logger)
-	handle(router, "/debug/pprof/profile", pprof.Profile, logger)
-	handle(router, "/debug/pprof/symbol", pprof.Symbol, logger)
+	// Debug endpoints
+	handle(router, "/debug/pprof", []string{"GET"}, pprof.Index, logger)
+	handle(router, "/debug/pprof/cmdline", []string{"GET"}, pprof.Cmdline, logger)
+	handle(router, "/debug/pprof/profile", []string{"GET"}, pprof.Profile, logger)
+	handle(router, "/debug/pprof/symbol", []string{"GET"}, pprof.Symbol, logger)
 
-	handle(router, "/sync", apiServer.syncAll, logger)
-	handle(router, "/sync/{client}", apiServer.syncOne, logger)
-	handle(router, "/status", apiServer.status, logger)
+	// Sync endpoints
+	handle(router, "/sync", []string{"POST"}, apiServer.syncAll, logger)
+	handle(router, "/sync/{client}", []string{"POST"}, apiServer.syncOne, logger)
```

Comment:
Might be nicer if we add a `var POST = []string{"POST"}` (and GET), instead of repeating this constant